node/src/components/linear_chain/pending_signatures.rs

```diff
@@ -117,11 +123,13 @@ impl PendingSignatures {
     fn mark_bonded_helper(&mut self, public_key: PublicKey, block_hash: BlockHash) -> Option<()> {
         match self.pending_finality_signatures.entry(public_key.clone()) {
             Entry::Occupied(mut validator_sigs) => {
-                let sig = validator_sigs.get_mut().remove(&block_hash)?;
+                let sig = validator_sigs.get_mut().get(&block_hash)?;
                 if !sig.is_bonded() {
+                    let bonded_status = VerificationStatus::Bonded(sig.value().clone());
+                    drop(sig);
                     validator_sigs
                         .get_mut()
-                        .insert(block_hash, VerificationStatus::Bonded(sig.into_inner()));
```

Comment:
Why not just remove the `if`? And just do:
```rust
let sig = validator_sigs.get_mut().remove(&block_hash)?;
validator_sigs.get_mut().insert(block_hash, VerificationStatus::Bonded(sig.into_inner()));
Some(())
```

Reply:
Could also be done with another match on an `entry`, I guess, but that might just be simpler.

Reply:
Hmm, right. The `if` here makes sure we don't overwrite an already `Bonded` entry but this doesn't matter. Good idea.

Reply:
Done in https://github.com/CasperLabs/casper-node/pull/1291/commits/221f3fc2b1b4e8ec51126ce59e396e1b02c56435